File: springwebfluxtutorial/src/main/java/com/qiusuo/webfluxtutorial/repository/CustomBookRepository.java
package com.qiusuo.webfluxtutorial.repository

import com.qiusuo.webfluxtutorial.domain.Book;
import reactor.core.publisher.Flux;

public interface CustomBookRepository {
    Flux<Book> findAllBooks();

    Flux<Book> findAllBooksForUser(String userId);
}

File: springwebfluxtutorial/src/main/kotlin/com/qiusuo/webfluxtutorial/repository/CustomBookRepository.kt
package com.qiusuo.webfluxtutorial.repository

import com.qiusuo.webfluxtutorial.domain.Book
import reactor.core.publisher.Flux

interface CustomBookRepository {
    fun findAllBooks(): Flux<Book>
    fun findAllBooksForUser(userId: String): Flux<Book>
}